pkg/bgp/speaker/speaker.go

```diff
@@ -423,36 +423,54 @@ func convertEndpointSliceV1(in *slim_discover_v1.EndpointSlice) *metallbspr.Endp
 	}
 	out := new(metallbspr.Endpoints)
 	for _, ep := range in.Endpoints {
-		for _, addr := range ep.Addresses {
-			out.Ready = append(out.Ready, metallbspr.Endpoint{
-				IP:       addr,
-				NodeName: ep.NodeName,
-			})
+		if isConditionReadyForSliceV1(ep.Conditions) {
```

Comment:
nit: maybe invert the condition and save one level of indentation?

```suggestion
		if !isConditionReadyForSliceV1(ep.Conditions) {
		        continue
		}
```